pkg/aws/aws.go

```diff
@@ -491,20 +492,32 @@ func (c *SdkClient) defaultRouteTableForVpc(vpcId string) (*ec2v2types.RouteTabl
 }
 
 func (c *SdkClient) listAllInstancesAttribute(att cloudtrailv2types.LookupAttribute) ([]cloudtrailv2types.Event, error) {
+	logging.Infof("Looking for events of type: %s", *att.AttributeValue)
 	// We only look up events that are not older than 2 hours
 	since := time.Now().UTC().Add(time.Duration(-2) * time.Hour)
-	// We only look up 1000 events maximum
-	var maxResults int32 = 1000
+	// We will only capture this many events via pagination - looping till we
+	// exhaust 90 days of events might take *very* long in big accounts
+	// otherwise.
+	maxNumberEvents := 1000
+	var events []cloudtrailv2types.Event = make([]cloudtrailv2types.Event, 0)
 	in := &cloudtrailv2.LookupEventsInput{
 		LookupAttributes: []cloudtrailv2types.LookupAttribute{att},
-		MaxResults:       &maxResults,
 		StartTime:        &since,
 	}
-	out, err := c.CloudtrailClient.LookupEvents(context.TODO(), in)
-	if err != nil {
-		return nil, err
+	paginator := cloudtrailv2.NewLookupEventsPaginator(c.CloudtrailClient, in)
+	// FIXME: Decide if we should just always retrieve *all* events which could
+	// be wasteful
+	for paginator.HasMorePages() && len(events) < maxNumberEvents {
+		out, err := paginator.NextPage(context.TODO())
+		if err != nil {
+			return nil, err
+		}
+		for _, e := range out.Events {
+			logging.Infof("Event found: %s", *e.EventId)
```

Comment:
This might become very noisy. Suggest making this debug.

Reply:
Oh yep I only added those because it didn't work while testing until I realized RH-SRE events are always considered valid